Tests/SecureXPCTests/Encoder & Decoder/Round Trip/File Descriptor XPC Wrapper Tests.swift

```diff
@@ -0,0 +1,194 @@
+        }
+        server.start()
+        
+        let container = try await client.send(to: route)
+        let handle = container.wrappedValue
+        defer { try! handle.close() }
+        
+        XCTAssertEqual(pathForFileDescriptor(fileDescriptor: handle.fileDescriptor), currentPath())
+    }
+    
+    func testFileHandle_PropertyWrapper() async throws {
+        struct SecureDocument: Codable {
+            var securityLevel: Int
+            @FileHandleForXPC var document: FileHandle
+        }
+        
+        let server = XPCServer.makeAnonymous()
+        let client = XPCClient.forEndpoint(server.endpoint)
+        let route = XPCRoute.named("secure", "document")
+                            .withReplyType(SecureDocument.self)
+        server.registerRoute(route) {
+            SecureDocument(securityLevel: 5, document: FileHandle(forReadingAtPath: self.currentPath())!)
+        }
+        server.start()
+        
+        let document = try await client.send(to: route).document
+        defer { try! document.close() }
+        
+        XCTAssertEqual(pathForFileDescriptor(fileDescriptor: document.fileDescriptor), currentPath())
+    }
+    
+    // MARK: Darwin file descriptor
+    
+    func testDarwinFileDescriptor_DirectInit() async throws {
+        let server = XPCServer.makeAnonymous()
+        let client = XPCClient.forEndpoint(server.endpoint)
+        let route = XPCRoute.named("fd", "provider")
+                            .withReplyType(DarwinFileDescriptorForXPC.self)
+        server.registerRoute(route) {
+            DarwinFileDescriptorForXPC(wrappedValue: open(self.currentPath(), O_RDONLY))
+        }
+        server.start()
+        
+        let container = try await client.send(to: route)
+        let descriptor = container.wrappedValue
+        defer { close(descriptor) }
+        
+        XCTAssertEqual(pathForFileDescriptor(fileDescriptor: descriptor), currentPath())
+    }
+    
+    func testDarwinFileDescriptor_PropertyWrapper() async throws {
+        struct SecureDocument: Codable {
+            var securityLevel: Int
+            @DarwinFileDescriptorForXPC var document: Int32
+        }
+        
+        let server = XPCServer.makeAnonymous()
+        let client = XPCClient.forEndpoint(server.endpoint)
+        let route = XPCRoute.named("secure", "document")
+                            .withReplyType(SecureDocument.self)
+        server.registerRoute(route) {
+            SecureDocument(securityLevel: 5, document: open(self.currentPath(), O_RDONLY))
+        }
+        server.start()
+        
+        let document = try await client.send(to: route).document
+        defer { close(document) }
+        
+        XCTAssertEqual(pathForFileDescriptor(fileDescriptor: document), currentPath())
+    }
+    
+    // MARK: Automatic bridging
+    
+    func testAutomaticBridging_DirectInit() async throws {
+        let server = XPCServer.makeAnonymous()
```

Comment:
These anonymous services are coming in really clutch for testing! I'm so glad I ran into that one random forum post that suggested it lol